### PR TITLE
[DOCFIX] [branch-1.6] Add S3 API docs with examples

### PR DESCRIPTION
**Pull request by ifcharming**
Here are screenshots:

screen shot 2017-08-29 at 4 13 06 pm
screen shot 2017-08-29 at 4 13 13 pm
screen shot 2017-08-29 at 4 13 21 pm
screen shot 2017-08-29 at 4 13 28 pm
screen shot 2017-08-29 at 4 13 37 pm
screen shot 2017-08-29 at 4 13 47 pm

### CODE REVIEW

File: docs/en/Clients-S3.md [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+  <tr>
+    <td>{{ item.S3Feature }}</td>
+    <td>{{ item.Status }}</td>
+  </tr>
+{% endfor %}
+</table>
+
+# Language support
+Alluxio S3 client supports various programming languages, such as C++, Java, Python, Golang, Ruby and etc.
+In this documentation, we use curl Rest calls and python S3 client as usage examples.
```
Comment (madanadit) [inline docs/en/Clients-S3.md]: capitalize REST

File: docs/en/Clients-S3.md [hunk 2]
```diff
@@ -0,0 +1,243 @@
...
+
+# Example Usage
+
+## Rest API
```
Comment (madanadit) [inline docs/en/Clients-S3.md]: throughout this doc

File: docs/en/Clients-S3.md [hunk 3]
```diff
@@ -0,0 +1,243 @@
...
+
+Alluxio supports a RESTFul API that is compatible with the basic operations of the Amazon 
+[S3 API](http://docs.aws.amazon.com/AmazonS3/latest/API/Welcome.html).
+
+The [REST API documentation](http://www.alluxio.org/restdoc/{{site.ALLUXIO_MAJOR_VERSION}}/proxy/index.html)
+is generated as part of Alluxio build and accessible through
+`${ALLUXIO_HOME}/core/server/proxy/target/miredot/index.html`.
+
+There are performance implications of using the HTTP proxy. In particular, using the proxy requires
+an extra hop. For optimal performance, it is recommended to run the proxy server and an Alluxio
```
Comment (madanadit) [inline docs/en/Clients-S3.md]: maybe include a line about load balancing when using multiple proxy servers

**Comment by madanadit** [comment]
Thanks @ifcharming. Left some minor comments

File: docs/en/Clients-S3.md (hunk 3 above)
Comment (ifcharming) [reply docs/en/Clients-S3.md]: done, thanks

File: docs/en/Clients-S3.md (hunk 1 above)
Comment (ifcharming) [reply docs/en/Clients-S3.md]: done

File: docs/en/Clients-S3.md (hunk 2 above)
Comment (ifcharming) [reply docs/en/Clients-S3.md]: updated all

**Comment by ifcharming** [comment]
@madanadit thanks for the comments! I've addressed your comments. PTAL

**Pull request merged**